### PR TITLE
Task#4 - Feat: Create Card component 

### LINKED ISSUE #11: Task#4 - DevTask: Create Card component
List comon components: [Link](https://docs.google.com/spreadsheets/d/1sGnQiUdub6pJTCG6TqeLHc4331FQE1Ub/edit#gid=2139907494)
## OVERVIEW
- Create common components: Card
   - CardHeader
   - CardContent
   - CardImg
   - CardFooter
- Create props for Card 
   - maxW,
   - name,
   - size,
   - variant,
   - onChange,
   - children, 
- Testing
### DEMO
image

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Closes #11 
## OVERVIEW
- Updated new Rule: Using const for class
- Create component Card:
   - Card class
   - Card:
      - CardHeader
      - CardContent
      - CardImage
      - CardFooter
   - Card props:
      - Src
      - size
      - ...
- Create productCard
```javascript
    <MyCard
      headerText=""
      imgSrc=".svg"
      title=""
      author=""
      date=""
    />
``` 
- Create serviceCard
```javascript
      <ServiceCard
        headerText=""
        title=""
        author=" "
        date=""
      />
``` 
- CraatehorizontalCard
```javascript
      <HorizontalCard
        title=""
        date=""
        author=""
        headerText=""
      />
``` 
- Testing

### DEMO TESTING
- produtCard
image
- serviceCard
image

- horizontalCard
image

### CODE REVIEW

File: practice-one/src/Components/Common/productCard.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Card from './Card';
+
+const MyCard = ({
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/productCard.jsx]: what is MyCard?

File: practice-one/src/Components/Common/productCard.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    </Card.CardHeader>
+    <Card.CardImage src={imgSrc} alt="Here is a photo of the product title above" />
+    <Card.CardContent>
+      <h3 className="pb-1.5 pt-1">
+        <a
+          href="#"
+          className="text-xl font-bold text-black-500"
+        >
+          {title}
+          {' '}
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/productCard.jsx]: what is this

File: practice-one/src/Components/Common/productCard.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        </a>
+      </h3>
+    </Card.CardContent>
+    <Card.CardFooter>
+      <h4>
+        {date}
+        ,
+        {' '}
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/productCard.jsx]: what is this

File: practice-one/src/Components/Common/productCard.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        <span className="text-gray-500">{author}</span>
+        {' '}
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/productCard.jsx]: The same here

File: practice-one/src/Components/Common/serviceCard.jsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      <Card.CardContent>
+        <div className="w-11/12 mb-6">
+          <h1 className="text-2xl font-bold mb-8 mt-3">{title}</h1>
+        </div>
+      </Card.CardContent>
+      <Card.CardFooter>
+        <h4>
+          {date}
+          ,
+          {' '}
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/serviceCard.jsx]: The same here

File: practice-one/src/Components/Common/Card.jsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    'secondary',
+    'success',
+    'danger',
+    'warning',
+  ]),
+  onChange: PropTypes.func,
+  children: PropTypes.node.isRequired,
+};
+
+const CardHeader = ({ children }) => (
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Card.jsx:50]: Missing proptypes

File: practice-one/src/Components/Common/Card.jsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  <div className="card-header bg-transparent border-none p-0 relative">{children}</div>
+);
+
+const CardContent = ({ children }) => (
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Card.jsx:61]: Missing proptypes

File: practice-one/src/Components/Common/Card.jsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  <div className="card-body bg-transparent border-none p-0 relative">{children}</div>
+);
+
+const CardFooter = ({ children }) => (
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Card.jsx:72]: Missing proptypes

File: practice-one/src/Components/Common/Card.jsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  <div className="card-footer bg-transparent border-none p-0 relative">{children}</div>
+);
+
+const CardImage = ({ src, alt }) => (
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Card.jsx:83]: Missing proptypes

File: practice-one/src/Components/Common/horizontalCard.jsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    <div className="flex justify-between items-end border-t-2 border-t-gray-500">
+      <div className="card-horizontal-left pt-5">
+        <Card.CardContent>
+          <h3 className="text-xl font-bold mb-6">{title}</h3>
+        </Card.CardContent>
+        <Card.CardFooter>
+          <h4>
+            {date}
+            ,
+            {' '}
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/horizontalCard.jsx]: The same here

File: practice-one/src/Components/Common/horizontalCard.jsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  <Card variant="primary" size="medium">
+    <div className="flex justify-between items-end border-t-2 border-t-gray-500">
+      <div className="card-horizontal-left pt-5">
+        <Card.CardContent>
+          <h3 className="text-xl font-bold mb-6">{title}</h3>
+        </Card.CardContent>
+        <Card.CardFooter>
+          <h4>
+            {date}
+            ,
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/horizontalCard.jsx]: use HTML char code

File: practice-one/src/Components/Common/Card.jsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+    success: 'border-green-500',
+    danger: 'border-red-500',
+    warning: 'border-yellow-500',
+  };
+
+  const cardClasses = `${commonClasses} ${sizeClasses[size]} ${variantClasses[variant]}`;
+
+  return (
+    <div className={`max-w-${maxW} mx-auto ${cardClasses}`} onChange={onChange}>
+      {name && <h2 className="text-sm font-bold mb-4">{name}</h2>}
```
Comment (phattruong-agilityio) [inline practice-one/src/Components/Common/Card.jsx]: I don't see any example using name. Please help check

**Review by phattruong-agilityio — APPROVED** [review]

File: practice-one/src/Components/Common/productCard.jsx (hunk 1 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/productCard.jsx]: Done, change name MyCard into ProductCard

File: practice-one/src/Components/Common/productCard.jsx (hunk 2 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/productCard.jsx]: Done,removed this

File: practice-one/src/Components/Common/productCard.jsx (hunk 3 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/productCard.jsx]: Done,removed this !

File: practice-one/src/Components/Common/productCard.jsx (hunk 4 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/productCard.jsx]: Done,removed this

File: practice-one/src/Components/Common/serviceCard.jsx (hunk 5 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/serviceCard.jsx]: Done,removed this

File: practice-one/src/Components/Common/Card.jsx (hunk 6 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/Card.jsx:50]: Yes, Added proptypes and commented line for variables

File: practice-one/src/Components/Common/Card.jsx (hunk 7 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/Card.jsx:61]: Yes, Added proptypes and commented line for variables

File: practice-one/src/Components/Common/Card.jsx (hunk 8 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/Card.jsx:72]: Yes, Added proptypes and commented line for variables

File: practice-one/src/Components/Common/Card.jsx (hunk 9 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/Card.jsx:83]: Yes, Added proptypes and commented line for variables

File: practice-one/src/Components/Common/horizontalCard.jsx (hunk 10 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/horizontalCard.jsx]: Done, removed this

File: practice-one/src/Components/Common/horizontalCard.jsx (hunk 11 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/horizontalCard.jsx]: Done,  used {String.fromCharCode(44)} for `,`

File: practice-one/src/Components/Common/Card.jsx (hunk 12 above)
Comment (TheHienDuong) [reply practice-one/src/Components/Common/Card.jsx]: Done, I removed this

**Review by phattruong-agilityio — APPROVED** [review]

**Pull request merged**